src/ray/core_worker/core_worker.cc

```diff
@@ -1641,13 +1641,18 @@ void CoreWorker::SubmitTask(const RayFunction &function,
       task_options.override_environment_variables;
   override_environment_variables.insert(current_override_environment_variables.begin(),
                                         current_override_environment_variables.end());
+  int64_t depth;
```

Comment:
It would be good if we could avoid putting a lock around depth, since there's only one thread accessing it anyway.

Can we move it to the WorkerContext, which is already tracking current task index, etc?